tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleCore.java

```diff
@@ -154,9 +159,48 @@ private static void configureIfLiferayProject( final IProject project, final Gra
         {
             Job job = new WorkspaceJob( "Checking gradle configuration" )
             {
+
                 @Override
                 public IStatus runInWorkspace( IProgressMonitor monitor ) throws CoreException
                 {
+                    IFile bndFile = project.getFile( "bnd.bnd" );
+
+                    //case 1: has bnd file
+                    if( bndFile != null && bndFile.exists() )
```

Comment:
sorry . Greg,  it is hard to extract the checks out of a job . cause  I use a var needAddNaure outside job , and I need to change the value of the var , but when I want to read the var in job, it need the var to be final. So I think it doesn't matter if checks inside or outside the job.